src/cargo/core/compiler/compilation.rs

```diff
@@ -97,8 +101,8 @@ impl<'cfg> Compilation<'cfg> {
             root_output: PathBuf::from("/"),
             deps_output: PathBuf::from("/"),
             host_deps_output: PathBuf::from("/"),
-            host_dylib_path: bcx.host_info.sysroot_libdir.clone(),
-            target_dylib_path: bcx.target_info.sysroot_libdir.clone(),
+            host_dylib_path: bcx.info(CompileKind::Host).sysroot_libdir.clone(),
+            target_dylib_path: bcx.info(default_kind).sysroot_libdir.clone(),
```

Comment:
In a world were cargo supports multiple targets, wouldn't this need to be a map?

Reply:
It would, yeah. `Compilation` would need some larger refactorings to support multi-target builds for sure. I'm thinking those would probably want to happen if we were to ever expose this at the CLI level, but before then it may be premature to apply the refactorings here. I don't mind going either way though!

Reply:
No, I just wanted to make sure I was understanding it correctly.